backend/app/services/rag/splitter/structural_semantic.py

```
@@ -0,0 +1,517 @@
+                    chunk.strip()
+                    for chunk in response.split("===CHUNK_BOUNDARY===")
+                    if chunk.strip()
+                ]
+                if chunks:
+                    return chunks
+
+            # Fallback to force splitting if LLM response is not useful
+            logger.warning("LLM response not useful, falling back to force splitting")
+            return [chunk for chunk, _ in self.force_split_with_overlap(text)]
+
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)